### PR TITLE
Propagate /no-warnings flag

### PR DESCRIPTION
**Pull request by vcsharma**
Fixes a bug where `/no-warnings` flag was not suppressing warnings.

### CODE REVIEW

**Comment by dmpots** [comment]
Change looks good, but you need to add a unit test.

**Comment by vcsharma** [comment]
@dmpots done!

**Comment by vcsharma** [comment]
Added a couple of tests to verify that warnings are not shown when `/no-warnings` flag is used and they get printed when `/no-warnings` flag is not used.  The exact behavior is mirrored when a user decides to  output warnings and errors to an output file by using /Fe flag. The unit tests only cover the stdout scenarios though.

**Comment by vcsharma** [comment]
Thanks @hekota! And yes, will do squash merge from this time onwards. ;)

File: lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp [hunk 1]
```diff
@@ -7257,8 +7257,7 @@ void ResourceToHandle::ReplaceResourceWithHandle(Value *ResPtr,
       // Remove resource Store.
       SI->eraseFromParent();
     } else {
-      if (!U->user_empty() || !isa<GEPOperator>(U))
-        DXASSERT(0, "invalid operation on resource");
+      DXASSERT(U->user_empty() && isa<GEPOperator>(U), "invalid operation on resource");
```
[the remark this answers is not in the document]
Comment (vcsharma) [reply lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp]: No. I just did "git pull" which pulled in some of the latest checkins. I just had a chat with @hekota on this. I would avoid doing git pulls on github repos.

File: tools/clang/test/CodeGenHLSL/check_nowarning.hlsl [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// RUN: %dxc -E main -no-warnings -T ps_6_0 %s | FileCheck %s
```
[the remark this answers is not in the document]
Comment (vcsharma) [reply tools/clang/test/CodeGenHLSL/check_nowarning.hlsl]: That was exactly my thought! I wanted to use -check-prefix to define multiple targets in one file and therefore avoid having multiple test files. I tried -check-prefix/--check-prefix, it didn't work.

File: lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (hekota) [reply lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp]: I forgot to mention that you can also do "git pull" and then "git rebase -i origin/master" to sync your branch to the latest code and avoid the other commits from showing up in your PR.

**Pull request merged**